### PR TITLE
Improve naming of commands and use AsCommand Attribute

### PR DESCRIPTION
**Pull request by aryaantony92**
Related to https://github.com/pimcore/pimcore/issues/15421

### CODE REVIEW

File: src/Command/IndexService/ResetQueueCommand.php [hunk 1]
```diff
@@ -47,6 +51,10 @@ protected function configure(): void
      */
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
+        // Remove in version 2.0
+        if($input->getArguments()['command'] == 'ecommerce:indexservice:reset-queue') {
+            $output->writeln('The command "portal-engine:update:index-recreate" is deprecated and will be removed in Version 2.0. Please use "ecommerce:index:reset-indices" instead.');
```
Comment (jdreesen) [inline src/Command/IndexService/ResetQueueCommand.php]:
```suggestion
            $output->writeln('The command "ecommerce:indexservice:reset-queue" is deprecated and will be removed in Version 2.0. Please use "ecommerce:index:reset-indices" instead.');
```

File: src/Command/IndexService/BootstrapCommand.php [hunk 2]
```diff
@@ -22,13 +22,19 @@
...
+use Symfony\Component\Console\Attribute\AsCommand;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
 
 /**
  * @internal
  */
+#[AsCommand(
+    name: 'ecommerce:index:create-or-update-indices',
```
Comment (kingjia90) [inline src/Command/IndexService/BootstrapCommand.php:34]: - [ ] Need to change the docs to use the replacement 
https://github.com/pimcore/ecommerce-framework-bundle/edit/1.0/doc/05_Index_Service/01_Product_Index_Configuration/05_Data_Architecture_and_Indexing_Process.md

About the naming, `indices` is fine, but would be `indexes` (both are grammatically fine) easier to read for non-natives? and could be just avoided because the part `:index:` is already hinting the domain/area of application?
in short, should it be  `ecommerce:index:create-or-update`?

File: src/Command/IndexService/ResetQueueCommand.php [hunk 3]
```diff
@@ -26,6 +27,11 @@
 /**
  * @internal
  */
+#[AsCommand(
+    name: 'ecommerce:index:reset-indices',
```
Comment (kingjia90) [inline src/Command/IndexService/ResetQueueCommand.php:31]:
```suggestion
    name: 'ecommerce:index:reset',
```
?
same as above (docs)

File: src/Command/IndexService/BootstrapCommand.php (hunk 2 above)
Comment (aryaantony92) [reply src/Command/IndexService/BootstrapCommand.php:34]: Thanks for the suggestion. Yes, we can rename it to 'ecommerce:index:create-or-update'. Should we also consider renaming the options https://github.com/pimcore/ecommerce-framework-bundle/blob/a576c8470d005df9312696a602d7f9a6ded33f46/src/Command/IndexService/BootstrapCommand.php#L65

Comment (kingjia90) [reply src/Command/IndexService/BootstrapCommand.php:34]: when possible yes, i'd say

File: src/Command/IndexService/BootstrapCommand.php [hunk 4]
```diff
@@ -22,13 +22,19 @@
...
+    description: 'Bootstrap tasks creating/updating index (for all tenants), use one of the options --create-or-update-index-structure or --update-index',
+    aliases: ['ecommerce:indexservice:bootstrap']
+)]
 class BootstrapCommand extends AbstractIndexServiceCommand
```
Comment (kingjia90) [inline src/Command/IndexService/BootstrapCommand.php:38]: just leaving a placeholder to remind that if command `ecommerce:indexservice:bootstrap` have changed, may require a filename and class name change too (`BoostrapCommand`)

**Comment by markus-moser** [comment]
Let's close this, the downsides are bigger then the advantages - see https://github.com/pimcore/pimcore/issues/15421

**Comment by markus-moser** [comment]
We could do the `AsCommand` part in a separate PR if someone has time to do it...
